### PR TITLE
Updates Princess shield mana reduction bonus to be %-based

### LINKED ISSUE #735: Possible inconsistencies across faction shield perks
The three factions' shields offer bonuses based on your faction rank (a scale of 1-10). Until #698, the first 3 levels of the Princess rank increased mana costs rather than reduced them. That PR addressed that issue by bounding the resulting value and keeping it from going negative, but also means that the first 3 levels of the Princess faction offer little to no mana cost reduction.

These are where the codebase's calculations would currently fall. Are these where we think they should be?

| piFactionRank | Mana Cost Reduction (Cess) | Added Spell Power (Duke) | Damage Reduction (Jonas) |
|---------------|------------|-----------|-----------------|
| 1             | 0          | 1         | 1               |
| 2             | 0          | 3         | 1               |
| 3             | 0          | 4         | 2               |
| 4             | 0          | 6         | 2               |
| 5             | 0          | 7         | 2               |
| 6             | 1          | 9         | 3               |
| 7             | 1          | 10        | 3               |
| 8             | 1          | 12        | 3               |
| 9             | 2          | 13        | 4               |
| 10            | 2          | 15        | 4               |

### PR DESCRIPTION
**Pull request by davidklaw**
This PR modifies the Princess faction's shield bonus (mana reduction) to be a percentage based on rank, ranging from a 5-25% reduction in mana costs. As seen in #735, the shield bonus for Princess soldiers is poor in comparison to the other factions; many of her faction levels offer NO bonus. Currently, the bonus maxes out at a savings of -2 mana per cast. With this change, the savings are a percentage of the cost and are detailed below.

I'd like to note that my formulas are maybe more verbose than they could be, but I thought it was important to show the values involved.

### Before and After Comparison

| piFactionRank | Mana Cost Reduction (Current) | Mana Cost Reduction (Proposed) |
|---------------|------------|-----------|
| 1             | 0          | 5%         |
| 2             | 0          | 7%         |
| 3             | 0          | 9%         |
| 4             | 0          | 11%         |
| 5             | 0          | 13%         |
| 6             | 1          | 16%         |
| 7             | 1          | 18%        |
| 8             | 1          | 20%        |
| 9             | 2          | 22%        |
| 10            | 2          | 25%        |

### Mana Cost = 5
| Faction Rank | Mana Cost | Calculated Mana Cost |
|--------------|-----------|----------------------|
|      1       |     5     |           5          |
|      2       |     5     |           5          |
|      3       |     5     |           5          |
|      4       |     5     |           5          |
|      5       |     5     |           5          |
|      6       |     5     |           5          |
|      7       |     5     |           5          |
|      8       |     5     |           4          |
|      9       |     5     |           4          |
|     10       |     5     |           4          |

### Mana Cost = 10
| Faction Rank | Mana Cost | Calculated Mana Cost |
|--------------|-----------|----------------------|
|      1       |    10     |           10          |
|      2       |    10     |           10          |
|      3       |    10     |           10          |
|      4       |    10     |           9          |
|      5       |    10     |           9          |
|      6       |    10     |           9          |
|      7       |    10     |           9          |
|      8       |    10     |           8          |
|      9       |    10     |           8          |
|     10       |    10     |           8          |

### Mana Cost = 20
| Faction Rank | Mana Cost | Calculated Mana Cost |
|--------------|-----------|----------------------|
|      1       |    20     |          19          |
|      2       |    20     |          19          |
|      3       |    20     |          19          |
|      4       |    20     |          18          |
|      5       |    20     |          18          |
|      6       |    20     |          17          |
|      7       |    20     |          17          |
|      8       |    20     |          16          |
|      9       |    20     |          16          |
|     10       |    20     |          15          |

### Mana Cost = 30

| Faction Rank | Mana Cost | Calculated Mana Cost |
|--------------|-----------|----------------------|
|      1       |    30     |          29          |
|      2       |    30     |          28          |
|      3       |    30     |          28          |
|      4       |    30     |          27          |
|      5       |    30     |          27          |
|      6       |    30     |          26          |
|      7       |    30     |          25          |
|      8       |    30     |          24          |
|      9       |    30     |          24          |
|     10       |    30     |          23          |

Closes #735

### CODE REVIEW

**Comment by paul-bargerstock** [comment]
This is a GREAT change.  Hats off for bringing 'Cess in line with her competition.

File: kod/object/item/passitem/defmod/shield/soldshld/prinshld.kod [hunk 1]
```diff
@@ -150,10 +158,30 @@ messages:
...
    GetSpellManaReduction()
+   "Gets the % by which mana is reduced for Princess shield holders."
    {
       local iReduction;
 
-      iReduction = Bound(((piFactionRank/3) - 1),0,$);
+      % Sample numbers for:
+      % - FACTION_RANK_MIN = 1
+      % - FACTION_RANK_MAX = 10
+      % - MANA_REDUCTION_MAX = 5
```
Comment (akirmse) [inline kod/object/item/passitem/defmod/shield/soldshld/prinshld.kod:168]: MIN, yes?

File: kod/object/item/passitem/defmod/shield/soldshld/prinshld.kod [hunk 2]
```diff
@@ -150,10 +158,30 @@ messages:
...
+      % -------------|---------------
+      %       1      |      5
+      %       2      |      6
+      %       3      |      7
+      %       4      |      9
+      %       5      |     10
+      %       6      |     12
+      %       7      |     14
+      %       8      |     17
+      %       9      |     19
```
Comment (akirmse) [inline kod/object/item/passitem/defmod/shield/soldshld/prinshld.kod:181]: This table is non-linear here, which doesn't match the formula.  Double check?

File: kod/object/passive/spell.kod [hunk 3]
```diff
@@ -1571,11 +1571,11 @@ messages:
          iCost = iCost - 1;
       }
 
-      % Up to -2 for being a Princess soldier
+      % Recieve a cost reduction for being a Princess soldier (Details in `prinshld.kod`)
```
Comment (akirmse) [inline kod/object/passive/spell.kod:1574]: "Receive"

**Comment by davidklaw** [comment]
Sorry, those were some sloppy mistakes. I've updated the tables here in the PR as well.

**Pull request merged**